Refresh AI landing and add new form

mtruj013 opened the pull request:
## Done

- Bumped Vanilla version to 4.1 
- Applied content and design changes to landing page as per [doc](https://docs.google.com/document/d/1HP9YdHF11yVQ9SLl0VXhLezDZREsr1GHIKNi9wT1Gv0/edit) and [mock-up](https://www.figma.com/file/sGeCjFvA1kV9Kmvg2S2VBn/23.10-U.com---AI-bubble?node-id=248%3A1446&mode=dev)
- Built new form in the new style as per [copy doc](https://docs.google.com/document/d/1maC3tCmTAgY1G_0nd0aAzw5ZMNeZqbkyawCRMG5Rgxk/edit) and [mock-up](https://www.figma.com/file/sGeCjFvA1kV9Kmvg2S2VBn/23.10-U.com---AI-bubble?node-id=764%3A9597&mode=dev)

## QA

- View the site locally in your web browser at: https://ubuntu-com-13138.demos.haus/ai
- Check against design and doc
- Make sure all CTAs work as expected
- Fill out the form and make sure it submits to marketo

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-5577, https://warthogs.atlassian.net/browse/WD-5547

webteam-app commented:
Demo starting at https://ubuntu-com-13138.demos.haus

mtruj013 commented:
@lyubomir-popov continuing from the discussion [here](https://warthogs.atlassian.net/browse/WD-5873?focusedCommentId=308884), I marked this page as ready for design review within the last 20 minutes, but I see your comments are from about an hour ago. If that's correct then you reviewed a WIP version of this page, I think the bulk of your comments have been addressed if you could take a look and confirm?

juanruitina commented:
- [ ] Logos should fit in a row, but I guess this has to do with the whole logo cloud kerfuffle.
- [ ] Fix the capitalisation here: "Read more about Ubuntu workstations ›"
- [ ] The spacing between the HR and the link at the bottom of the grid of Charmed apps (e.g. What is Kubeflow ›" is too tight. I think just wrapping the link on a `p` element would do?
- [ ] Also here, can we please increase the space between for each Charmed app on mobile, or remove the spacing between paragraph and HR+link? It gets confusing there, where one ends and the other starts. Maybe @lyubomir-popov can assist with this.

![Screenshot 2023-09-07 at 13 22 46](https://github.com/canonical/ubuntu.com/assets/1409729/d562dd13-5490-45a4-bc17-7ece129b547c)

- [ ] The image contain doesn't align with the grid, it's wider. Also, the gray background goes beyond the illustration, at least at the bottom.
- [ ] Please remove chevron after "Case study: Real-time space tracking with AI supercomputing"

lyubomir-popov commented:
One more, sorry! The links here need to be in paragraphs - so theyu get the padding-top/margin-bottom 
![image](https://github.com/canonical/ubuntu.com/assets/2741678/05039dd5-170e-4bf5-b0b6-cfd1404e6e9f)

lyubomir-popov commented:
@mtruj013 re-reviewing, can you please make the logos 128px (8rem) tall:
![image](https://github.com/canonical/ubuntu.com/assets/2741678/ae70024b-67a7-4548-a80f-58fc544f611e)

this should be top-aligned:
![image](https://github.com/canonical/ubuntu.com/assets/2741678/762366fc-7e78-4fa5-8c46-c26b8c1ebd9d)

ca nwe please avoid this gap:
![image](https://github.com/canonical/ubuntu.com/assets/2741678/a91a9be0-1ad4-43b6-a410-0d90e78c5d2e)

The spacing around this (top/bottom) is a bit much, ca nwe please have u-no-margin--bottom on the buttons, u-no-padding--bottom on all strips/sections.

mtruj013 commented:
@juanruitina @lyubomir-popov thanks both! I've addressed your comments if you could take another look?

Also about the Canonical + Nvidia logo being top aligned, the extra spacing on the top comes from the asset itself, is another version that can be used? 
![image](https://github.com/canonical/ubuntu.com/assets/17607612/15dbfa01-48ef-4dfe-8677-a0f0a3f99c39)

juanruitina commented:
Great job, LGTM!

lyubomir-popov commented:
@mtruj013 some tiny spacing things still appear in the pr:

- buttons can't be in p's - it interferes with the spacing:

![image](https://github.com/canonical/ubuntu.com/assets/2741678/85d7a296-1b7f-431e-b2ed-1182d053d99c)

- this p-section class is not needed - we need to get rid of this padding:
- 
![image](https://github.com/canonical/ubuntu.com/assets/2741678/2f354a4b-6b74-488c-913f-bee641541890)

- p-section still has u-no-margin, instead of u-no-padding--bottom, so the padding is still there:
(together, these 3 things add about 100px of extra white space and push content unneceserily) 
![image](https://github.com/canonical/ubuntu.com/assets/2741678/c4595c57-cf4b-4972-8a6d-84d379b4c396)

- small overlap here on medium screens:
![image](https://github.com/canonical/ubuntu.com/assets/2741678/918c0f0f-56b9-453a-a2a3-627e86b21b84)

- the orange tag still not attached to the hr above it:
![image](https://github.com/canonical/ubuntu.com/assets/2741678/a5960fef-ee66-4fdc-b2cd-e9c45412667f)

The rest looks really nice, thank you for the hard work on this.

mtruj013 commented:
Thanks @lyubomir-popov, I'll add the fixes, but for the Canonical + Nivida logo please see my [comment above](https://github.com/canonical/ubuntu.com/pull/13138#issuecomment-1711704505)

mtruj013 commented:
@lyubomir-popov that spacing distorts the image a bit
![image](https://github.com/canonical/ubuntu.com/assets/17607612/6ecc5bc3-5f80-4e63-8f07-4247dfbe94ea)
so I adjusted slightly 
![image](https://github.com/canonical/ubuntu.com/assets/17607612/2b2e11b1-8fc6-45ca-9ab6-52e58078e408)

anyway, ready for another look!

lyubomir-popov commented:
@mtruj013 you're right, 41 is thecorrect size!

juanruitina commented:
Forgot to QA the form, my apologies. Please make sure to add the semi-transparent overlay under it. I also think some more padding is needed around it, particularly the top one.

carkod commented on templates/shared/forms/interactive/ai.html:
```diff
@@ -0,0 +1,120 @@
+                <hr class="p-rule">
+                <label for="mlopsRequirements">Add details about your AI or MLOps requirements</label>
+                <textarea id="mlopsRequirements"></textarea>
+                
```
There is one line added by Sophie missing that says "Add your information"?

mtruj013 replied:
@Sophie-32 @juanruitina @lyubomir-popov in my opinion it looks better without that line. It looks a bit out of place. See:
![Screenshot from 2023-09-11 09-48-46](https://github.com/canonical/ubuntu.com/assets/17607612/636211e9-31fa-4f05-8469-7163763a13d4)
 
vs

![image](https://github.com/canonical/ubuntu.com/assets/17607612/c5ab9777-a454-45b1-821a-aa96e1eca6e0)


If we have to keep it, what's the ask from a design perspective (it wasn't included in the mock-up) to make it look a bit better? An hr on top? Lmk what you all think is best and I'll implement ASAP!

Sophie-32 replied:
I'd prefer to keep it, but will keep this in mind for future forms. I defer to @lyubomir-popov on adding the HR, but that seems like the best option from my perspective.

carkod commented on package.json:
```diff
@@ -115,7 +115,7 @@
     "url-polyfill": "1.1.12",
     "url-search-params-polyfill": "8.1.1",
-    "vanilla-framework": "^3.14.0",
+    "vanilla-framework": "4.1.0",
```
wow aren't there any breaking changes that could be happening with this?

mtruj013 replied:
We can't really rebrand bubbles while not being on V4. (I mean we can, but then we'd kind of have to do the work twice cause at some point all the custom styles I'd have to write to make it _look_ like it's on V4 would need to removed). But yes, regardless, this needs to be thoroughly QA'd. Relevant discussion [here](https://chat.canonical.com/canonical/pl/wi8kg469sjrp5r1utprbf4b9fo)

lyubomir-popov commented:
@mtruj013 the lines are there to indicate there are 3 subsecitons in the form, I'd rather keep them.

One thing - this px hardcoding doesn't work on larger screens - as the value of 1rem changes to 1.125 rem. It should use the $grid-max-width variable (or a rem based hardcoded equivalent)./ Otherwise, it becomes narrower than the grid on bigger screens (>1680):

![image](https://github.com/canonical/ubuntu.com/assets/2741678/18a1c8ae-bd4e-485b-bc95-5d7b875629e4)

mtruj013 commented:
>  the lines are there to indicate there are 3 subsecitons in the form, I'd rather keep them.

@lyubomir-popov I'm not asking about the lines, I'm asking about the extra line of copy "Add your information" that wasn't included in the design. From the mock-up: 
![image](https://github.com/canonical/ubuntu.com/assets/17607612/0966103f-950f-4ecc-b2c4-d4cdccb8a913)

from the doc (the pink bit is what's missing): 
![image](https://github.com/canonical/ubuntu.com/assets/17607612/2be44738-a9e6-44c6-a904-cc53b4dce118)


So my question was how to style this since it wasn't included in the mock-up. Should I just add an hr on top? It currently looks like this:
![image](https://github.com/canonical/ubuntu.com/assets/17607612/852cae82-0992-4f56-a765-4dbb087ecef1)

mtruj013 commented:
@lyubomir-popov I've updated the max-width variable, please let me know if this was what you meant

lyubomir-popov commented:
@mtruj013 I meant set the max-width (not min-width, this might break small screens) to this vanilla variable:
https://github.com/canonical/vanilla-framework/blob/eefce1c418b22c117ce8aeaf802dc58dceebf1eb/scss/_settings_grid.scss#L5

currently, it is even narower (875px):
![image](https://github.com/canonical/ubuntu.com/assets/2741678/c3a169e3-6b26-4a80-b9e4-33cea58aff4f)


you'll have to factor in the dialog padding and also poosition the close button absolutely - otherwise the grid in the dialog is still misaligned with the grid on the regular page:
![image](https://github.com/canonical/ubuntu.com/assets/2741678/85aa16f6-ea01-408e-b962-55d947a778a4)

mtruj013 commented:
Update: We've had a call and decided to merge as is, we will tackle any additional changes to the modal in a separate PR
